packages/amplify-category-analytics/src/utils/pinpoint-helper.ts

```diff
@@ -84,3 +84,43 @@ export const pinpointHasInAppMessagingPolicy = (context: $TSContext): boolean =>
   const { cfnTemplate } = readCFNTemplate(pinpointCloudFormationTemplatePath, { throwIfNotExist: false }) || {};
   return !!cfnTemplate?.Parameters?.[pinpointInAppMessagingPolicyName];
 };
+
+/**
+ * checks if notifications category has a pinpoint resource - legacy projects
+ */
+export const checkIfNotificationsCategoryHasPinpoint = (context: $TSContext): $TSAny => {
+  const { amplify } = context;
+  const { amplifyMeta } = amplify.getProjectDetails();
+  let pinpointApp: $TSAny;
+
+  if (amplifyMeta.notifications) {
+    const categoryResources = amplifyMeta.notifications;
+    Object.keys(categoryResources).forEach(resource => {
+      if (categoryResources[resource].service === AmplifySupportedService.PINPOINT && categoryResources[resource].output.Id) {
+        pinpointApp = {};
+        pinpointApp.appId = categoryResources[resource].output.Id;
+        pinpointApp.appName = resource;
+      }
+    });
+  }
+
+  return pinpointApp;
+};
+
+/**
+ * returns provider pinpoint region mapping
+ */
+export const getTemplateMappings = async (context: $TSContext): Promise<Record<string, $TSAny>> => {
```

Comment:
1. Do we have values that are of different type here ?
2. Should the method be `getPinpointRegionMappings` ?

Reply:
1. did not really investigate what types can it be as I just moved it in order to reuse in 2 different places
2. renamed